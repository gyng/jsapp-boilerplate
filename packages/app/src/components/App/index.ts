import { App } from "./App";

// Export as default to support webpack dynamic import
// This dynamic import is used for lazy-loading App
export default App;
