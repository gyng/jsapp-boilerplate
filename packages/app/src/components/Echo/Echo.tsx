import * as React from "react";

export const Echo = (
  props: { text?: string } = { text: "Default!" }
): JSX.Element => <p style={{ fontStyle: "italic" }}>{props.text}</p>;
