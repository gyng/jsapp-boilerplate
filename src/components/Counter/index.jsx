// @flow
/* eslint-disable react/prefer-stateless-function */

import React from "react";
import PropTypes from "prop-types";

export default class Counter extends React.Component {
  render() {
    return (
      <div>
        <div className="value">
          {this.props.value}
        </div>
        <button className="increment" onClick={this.props.onIncrementClick}>
          INCREMENT
        </button>
        <button className="decrement" onClick={this.props.onDecrementClick}>
          DECREMENT
        </button>
        <button
          className="increment"
          onClick={this.props.onIncrementClickAsync}
        >
          INCREMENT AFTER 1 SECOND
        </button>
      </div>
    );
  }
}

Counter.propTypes = {
  value: PropTypes.number.isRequired,
  onIncrementClick: PropTypes.func.isRequired,
  onIncrementClickAsync: PropTypes.func.isRequired,
  onDecrementClick: PropTypes.func.isRequired
};
