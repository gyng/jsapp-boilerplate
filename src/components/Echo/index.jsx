// @flow

import React from "react";

import s from "./styles.scss";

const Echo = (props: { text: string }) =>
  <p className={s.text}>
    {props.text}
  </p>;

export default Echo;
