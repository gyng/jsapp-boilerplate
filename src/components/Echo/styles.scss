.text {
  font-style: italic;
  text-align: center;
}
