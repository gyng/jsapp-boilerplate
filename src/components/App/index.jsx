// @flow
/* eslint-disable react/prefer-stateless-function, react/forbid-prop-types */

import React from "react";
import PropTypes from "prop-types";
import { Route, Link } from "react-router-dom";

import Counter from "containers/Counter";
import Echo from "components/Echo";

import hello from "./hello.jpg";
import s from "./styles.scss";

export default class App extends React.Component {
  static defaultProps: {
    className: string
  };

  render() {
    return (
      <div className={this.props.className}>
        <img className={s.robot} src={hello} alt="Cute robot?" />
        <Echo text="Hello, world! Find me in src/components/App/index.jsx!" />

        <div style={{ border: "solid 1px grey" }}>
          <Route
            exact
            path={this.props.match.url}
            render={() =>
              <Link to="/counter">
                Link to /counter. Click to show counter. Back/Forward buttons
                work.
              </Link>}
          />
          <Route path="/counter" component={Counter} />
        </div>
      </div>
    );
  }
}

App.propTypes = {
  className: PropTypes.string,
  match: PropTypes.object
};

App.defaultProps = {
  children: null,
  className: s.app,
  match: { url: "unknown" }
};
