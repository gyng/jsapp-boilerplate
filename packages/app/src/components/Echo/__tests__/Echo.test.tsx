import { mount } from "enzyme";
import * as React from "react";
import { Echo } from "@src/components/Echo";

describe("Echo", () => {
  it("renders the text", () => {
    const wrapper = mount(<Echo text="Hello, world!" />);
    const p = wrapper.find("p");
    expect(p).toHaveLength(1);
    expect(p.text()).toEqual("Hello, world!");
  });

  it("can do snapshot tests", () => {
    // You can perform snapshot testing if you so prefer
    const wrapper = mount(<Echo text="Hello, world!" />);
    expect(wrapper).toMatchSnapshot();
  });
});
