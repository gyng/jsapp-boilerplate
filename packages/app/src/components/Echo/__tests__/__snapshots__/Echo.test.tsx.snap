// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Echo can do snapshot tests 1`] = `ReactWrapper {}`;
